<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1440</width>
    <height>943</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="minimumSize">
    <size>
     <width>1440</width>
     <height>900</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>1440</width>
     <height>900</height>
    </size>
   </property>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>30</y>
      <width>1431</width>
      <height>771</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="mainContentLayout">
     <property name="spacing">
      <number>12</number>
     </property>
     <item>
      <layout class="QVBoxLayout" name="leftColumnLayout">
       <property name="spacing">
        <number>12</number>
       </property>
       <item>
        <widget class="QGroupBox" name="visualizationGroup">
         <property name="title">
          <string>üìä Robot Visualization</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QGraphicsView" name="graphicsView">
            <property name="minimumSize">
             <size>
              <width>550</width>
              <height>550</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>450</width>
              <height>450</height>
             </size>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="vizLabel">
            <property name="text">
             <string>2D Top-Down View ‚Ä¢ Real-time Forward Kinematics</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignmentFlag::AlignCenter</set>
            </property>
            <property name="subtitle" stdset="0">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="statusGroup">
         <property name="title">
          <string>üéØ Position &amp; Orientation</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <widget class="QLabel" name="labelTheta1">
            <property name="text">
             <string>Œò1: 0¬∞ (Feedback: 0¬∞)</string>
            </property>
            <property name="status" stdset="0">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="labelGripper">
            <property name="text">
             <string>Gripper: 0%</string>
            </property>
            <property name="status" stdset="0">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="labelOrientationZ">
            <property name="text">
             <string>Z: 0¬∞</string>
            </property>
            <property name="status" stdset="0">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="labelTheta2">
            <property name="text">
             <string>Œò2: 0¬∞ (Feedback: 0¬∞)</string>
            </property>
            <property name="status" stdset="0">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="middleColumnLayout">
       <property name="spacing">
        <number>12</number>
       </property>
       <item>
        <widget class="QGroupBox" name="serialGroup">
         <property name="title">
          <string>üîå Serial Connection</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="label_port">
            <property name="text">
             <string>COM Port:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="Com_port"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_baud">
            <property name="text">
             <string>Baud Rate:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="baudrate_port"/>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="QPushButton" name="Connect">
            <property name="text">
             <string>üîó Connect</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0" colspan="2">
           <widget class="QLineEdit" name="lineEdit">
            <property name="text">
             <string>üîå Disconnected</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignmentFlag::AlignCenter</set>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="positionControlGroup">
         <property name="title">
          <string>üéÆ XYZ Position Control</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="label_x">
            <property name="text">
             <string>X (mm):</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1" colspan="3">
           <widget class="QSpinBox" name="spinBox">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>40</height>
             </size>
            </property>
            <property name="minimum">
             <number>-1000</number>
            </property>
            <property name="maximum">
             <number>1000</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_y">
            <property name="text">
             <string>Y (mm):</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1" colspan="3">
           <widget class="QSpinBox" name="spinBox_2">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>40</height>
             </size>
            </property>
            <property name="minimum">
             <number>-1000</number>
            </property>
            <property name="maximum">
             <number>1000</number>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_z">
            <property name="text">
             <string>Z (mm):</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1" colspan="3">
           <widget class="QSpinBox" name="spinBox_3">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>40</height>
             </size>
            </property>
            <property name="minimum">
             <number>-100</number>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_gripper">
            <property name="text">
             <string>Gripper (%):</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1" colspan="3">
           <widget class="QSpinBox" name="spinBox_4">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>40</height>
             </size>
            </property>
            <property name="maximum">
             <number>360</number>
            </property>
           </widget>
          </item>
          <item row="4" column="0" colspan="4">
           <widget class="QLabel" name="label_manual">
            <property name="text">
             <string>Manual Control:</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignmentFlag::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QPushButton" name="pushButton_5">
            <property name="minimumSize">
             <size>
              <width>60</width>
              <height>60</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>60</width>
              <height>60</height>
             </size>
            </property>
            <property name="text">
             <string>‚ñ≤
Y+</string>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QPushButton" name="pushButton_2">
            <property name="minimumSize">
             <size>
              <width>60</width>
              <height>60</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>60</width>
              <height>60</height>
             </size>
            </property>
            <property name="text">
             <string>‚óÑ
X-</string>
            </property>
           </widget>
          </item>
          <item row="6" column="2">
           <widget class="QPushButton" name="pushButton_4">
            <property name="minimumSize">
             <size>
              <width>60</width>
              <height>60</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>60</width>
              <height>60</height>
             </size>
            </property>
            <property name="text">
             <string>‚ñ∫
X+</string>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <widget class="QPushButton" name="pushButton_3">
            <property name="minimumSize">
             <size>
              <width>60</width>
              <height>60</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>60</width>
              <height>60</height>
             </size>
            </property>
            <property name="text">
             <string>‚ñº
Y-</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QLabel" name="label_center">
            <property name="styleSheet">
             <string notr="true">font-size: 24px; color: #64748b;</string>
            </property>
            <property name="text">
             <string>‚äï</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignmentFlag::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QGroupBox" name="savedPositionsGroup">
       <property name="title">
        <string>üíæ Saved Positions</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QTableView" name="tableView">
          <property name="minimumSize">
           <size>
            <width>360</width>
            <height>0</height>
           </size>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::SelectionMode::SingleSelection</enum>
          </property>
          <property name="selectionBehavior">
           <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QPushButton" name="saveButton">
            <property name="text">
             <string>üíæ Save</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="loadButton">
            <property name="text">
             <string>üì• Load</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="runButton">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>50</height>
           </size>
          </property>
          <property name="text">
           <string>‚ñ∂Ô∏è RUN SEQUENCE</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_runInfo">
          <property name="text">
           <string>Save multiple positions, then press RUN to execute them sequentially with 5-second intervals.</string>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
          <property name="subtitle" stdset="0">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="rightColumnLayout">
       <property name="spacing">
        <number>12</number>
       </property>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget_2">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>367</width>
      <height>36</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="headerLayout">
     <property name="spacing">
      <number>2</number>
     </property>
     <item>
      <widget class="QLabel" name="titleLabel">
       <property name="text">
        <string>SCARA Robot Control System</string>
       </property>
       <property name="heading" stdset="0">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="subtitleLabel">
       <property name="text">
        <string>Selective Compliance Assembly Robot Arm - L1=232mm, L2=180mm</string>
       </property>
       <property name="subtitle" stdset="0">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1440</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
